react/features/dial-out/components/DialOutNumbersForm.web.js

```diff
@@ -0,0 +1,355 @@
+            dialInput: e.target.value
+        }, () => {
+            this._onDialNumberChange();
+        });
+    }
+
+    /**
+     * Updates the dialNumber when changes to the dial text or code happen.
+     *
+     * @private
+     * @returns {void}
+     */
+    _onDialNumberChange() {
+        const dialInput = this.state.dialInput;
+
+        const dialNumber
+            = this.state.selectedCode
+            ? this.state.selectedCode.content
+                + this._formatDialInput(dialInput)
+            : dialInput;
+
+        const formattedNumber = this._formatDialNumber(dialNumber);
+
+        this.props.onChange(formattedNumber);
+    }
+
+    /**
+     * Formats the dial input text.
+     *
+     * @param {string} dialInput - The text input for dial number.
+     * @private
+     * @returns {string} - The formatted phone number.
+     */
+    _formatDialInput(dialInput) {
+        let startIndex = 0;
+
+        if (dialInput.startsWith('00')) {
+            startIndex = 2;
+        } else if (dialInput.startsWith('0') || dialInput.startsWith('+')) {
+            startIndex = 1;
+        }
+
+        return dialInput.substring(startIndex);
+    }
+
+    /**
+     * Formats the dial number and country code as a whole.
+     *
+     * @param {string} dialNumber - The phone number to format.
+     * @private
+     * @returns {string} - The formatted phone number.
+     */
+    _formatDialNumber(dialNumber) {
+        let formattedNumber = dialNumber.replace(/\s+/g, '');
+
+        if (formattedNumber.startsWith('+')) {
+            formattedNumber = formattedNumber.substring(1);
+        }
+
+        return formattedNumber;
+    }
+
+    /**
+     * Sets the internal state to either open or close the dropdown. If the
+     * dropdown is disabled, the state will always be set to false.
+     *
+     * @param {Object} dropdownEvent - The even returned from clicking on the
+     * dropdown trigger.
+     * @private
+     * @returns {void}
+     */
+    _onOpenChange(dropdownEvent) {
+        this.setState({
+            isDropdownOpen: dropdownEvent.isOpen
+        });
+    }
+
+    /**
+     * Updates the internal state of the currently selected country code.
+     *
+     * @param {Object} selection - Event from choosing an dropdown option.
+     * @private
+     * @returns {void}
+     */
+    _onSelect(selection) {
+        this.setState({
+            isDropdownOpen: false,
+            selectedCode: selection.item
+        }, () => {
+            this._onDialNumberChange();
+        });
+    }
+
+    /**
+     * Updates the internal state of the currently selected number by defaulting
+     * to the first available number.
+     *
+     * @param {Object} countryCodes - The list of country codes to choose from
+     * for setting a default code.
+     * @private
+     * @returns {void}
+     */
+    _setDefaultCode(countryCodes) {
+        this.setState({
+            selectedCode: this._formatCountryCodes(countryCodes)[0]
```

Comment:
What do you think about the idea to have the `selectedCode` to be only `country` and not this complex structure from `_formatCountryCodes `? Or it can be even a simple `number` - id of the country or something like that. I think this will simplify the code and also it should optimize it.